src/App.tsx

```diff
@@ -1,14 +1,85 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
+import { Todo } from './types/Todo';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
+import { getTodos } from './api';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Filter } from './types/Filter';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [todosToShow, setTodosToShow] = useState<Todo[]>([]);
+  const [isLoadingTodos, setIsLoadingTodos] = useState<boolean>(false);
+  const [hasError, setHasError] = useState<boolean>(false);
+  const [activeTodo, setActiveTodo] = useState<Todo | null>(null);
+  const [filterBy, setFilterBy] = useState<Filter>(Filter.ALL);
+  const [query, setQuery] = useState<string>('');
+
+  const renderTodoList = () => {
+    if (!isLoadingTodos && todosToShow.length > 0) {
+      return (
+        <TodoList
+          todos={todosToShow}
+          activeTodo={activeTodo}
+          setActiveTodo={setActiveTodo}
+        />
+      );
+    }
+
+    if (hasError) {
+      return 'Error occured while loading todos!';
+    }
+
+    if (!isLoadingTodos && todosToShow.length === 0) {
```

Comment:
Replace `todosToShow.length === 0` with `!todosToShow.length`, it will be the same.